src/lib.rs

```diff
@@ -511,6 +524,10 @@ impl NotificationEmitter {
                 Err(e) => return Err(zbus::Error::MissingParameter(e)),
             };
         }
+        // marmarek: the above image should be processed to indicate from where the notification
+        // comes; something like overlaying qube icon in the right bottom icon corner; but that can
+        // come in the next iteration, but until then, ignore 'image-data' hint as that would
+        // override app icon (see the earlier comment)
```

Comment:
For simplicity, the current code only prefixes the notification title.  I’m not sure if this is sufficient — from what you have written here, I assume it isn’t.

Reply:
See the sketches posted on qubes matrix channel; generally, just prefixing the title is okay for the "experimental" (opt-in) state, but we need to do better before making it the default feature.